x-pack/plugins/monitoring/public/components/logstash/pipeline_viewer/views/config_viewer/plugin_statement.js

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import {
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+} from '@elastic/eui';
+import { formatMetric } from '../../../../../lib/format_number';
+import { Metric } from './metric';
+
+function getInputStatementMetrics({ latestEventsPerSecond }) {
+  return [
+    new Metric(
+      'eventsEmitted',
+      'cv-inputMetric__eventsEmitted',
+      formatMetric(latestEventsPerSecond, '0.[00]a', 'e/s emitted')
+    )
+  ];
+}
+
+function getProcessorStatementMetrics(processorVertex) {
+  const {
+    latestMillisPerEvent,
+    latestEventsPerSecond,
+    percentOfTotalProcessorTime,
+  } = processorVertex;
+
+  const cpuHighlight = processorVertex.isTimeConsuming() ? 'cv-processorMetric__cpuTimeHighlight' : '';
+  const eventMillisHighlight = processorVertex.isSlow() ? 'cv-processorMetric__eventMillis' : '';
+
+  return [
+    new Metric(
+      'cpuMetric',
+      `cv-processorMetric__cpuTime ${cpuHighlight}`,
+      formatMetric(Math.round(percentOfTotalProcessorTime || 0), '0', '%', { prependSpace: false })
+    ),
+    new Metric(
+      'eventMillis',
+      `cv-processorMetric__eventMillis ${eventMillisHighlight}`,
+      formatMetric(latestMillisPerEvent, '0.[00]a', 'ms/e')
+    ),
+    new Metric(
+      'eventsReceived',
+      'cv-processorMetric__events',
+      formatMetric(latestEventsPerSecond, '0.[00]a', 'e/s received')
+    )
+  ];
+}
+
+function renderPluginStatementMetrics(pluginType, vertex) {
+  const metrics = pluginType === 'input'
+    ? getInputStatementMetrics(vertex)
+    : getProcessorStatementMetrics(vertex);
+
+  return metrics.map(({ name, className, value }) => (
+    <EuiFlexItem
+      grow={false}
+      className={"cv-pluginStatement__metricContainer"}
+      key={name}
+    >
+      <div className={className}>
+        {value}
+      </div>
+    </EuiFlexItem>
+  ));
+}
+
+export function PluginStatement({
+  statement: {
+    hasExplicitId,
+    id,
+    name,
+    pluginType,
+    vertex
+  },
+  onShowVertexDetails
+}) {
```

Comment:
Given the relative complexity of the "shape" of the props here, consider adding `propTypes`. See https://github.com/yannickcr/eslint-plugin-react/issues/237#issuecomment-146210088 if you need some details on how, but note that that issue is nearly 3 years old so you might have to make some tweaks.